### PR TITLE
Bump dependencies to bitcoin 0.27 and bech32 0.8

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Plus various small no_std bits, working towards no_std.

### CODE REVIEW

File: lightning/src/lib.rs [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #![cfg_attr(all(any(test, feature = "_test_utils"), feature = "unstable"), feature(test))]
 #[cfg(all(any(test, feature = "_test_utils"), feature = "unstable"))] extern crate test;
 
+#[macro_use]
```
[the remark this answers is not in the document]
Comment (TheBlueMatt) [reply lightning/src/lib.rs:34]: Try adding a `#![cfg_attr(feature = no-std), no_std]`. Then this removes about 10% of the errors.

[the remark this answers is not in the document]
Comment (devrandom) [reply lightning/src/lib.rs:34]: ah, got it

**Review by devrandom — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -27,3 +27,6 @@ panic = "abort"
 opt-level = 3
 codegen-units = 1
 lto = true
+
+[patch.crates-io]
+core2 = { git = "https://github.com/TheBlueMatt/core2", branch = "main" }
```
Comment (jkczyz) [inline Cargo.toml:32]: Is this still `[DO NOT MERGE]`?

Comment (TheBlueMatt) [reply Cargo.toml:32]: Yea, depends on https://github.com/technocreatives/core2/pull/7. I *guess* we could just bump the CI MSRV version for no-std, but I don't see a reason to - gotta replace the `std::io` uses we have with `core2` anyway before we can ship no-std, so its not like we're in a rush.

**Review by jkczyz — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Dropped the `core2` patch since it seems like that isn't happening soon and just set the MSRV for no-std to stable instead for now. I think we can land this.

**Comment by devrandom** [comment]
ACK

**Comment by devrandom** [comment]
Interesting test failure though

**Comment by TheBlueMatt** [comment]
> Interesting test failure though

The test failure is unrelated. rustc is (probably) going to make an incompatible libstd change soon, so we're failing on beta. I've argued a bit with them, but there's not really a great other way to accomplish what they want (more `ErrorKind` variants).

**Comment by TheBlueMatt** [comment]
This hasn't materially changed since @jkczyz's last review, just merging.

**Pull request merged**